include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -271,8 +300,50 @@ namespace RINGMesh
 
         GEO::AttributesManager attributes_manager_;
     };
+    ALIAS_2D_AND_3D( CartesianGridBase );
+
+    template < index_t DIMENSION >
+    class CartesianGrid final : public CartesianGridBase< DIMENSION >
+    {
+        friend class CartesianGridBuilder< DIMENSION >;
+    };
     ALIAS_2D_AND_3D( CartesianGrid );
 
+    template <>
+    class CartesianGrid< 3 > final : public CartesianGridBase< 3 >
+    {
+        friend class CartesianGridBuilder< 3 >;
+
+    public:
+        CartesianGrid( ivec3 nb_cells_in_each_direction,
+            ReferenceFrame3D vec_cartesian_axis )
+            : CartesianGridBase(
+                  nb_cells_in_each_direction, vec_cartesian_axis ),
+              grid_cage_()
+        {
+            for( auto i : range( 3 ) )
+            {
+                vec3 origin1 = cartesian_frame_.origin(),
+                     vector1 = cartesian_frame_[i];
+                grid_cage_.push_back( Geometry::Plane{ vector1, origin1 } );
```

Comment:
Prefere emplacement when possible: grid_cage_.emplace_back( vector1, origin1 );
This will create your object directly into the vector instead of copying it into the vector.